flex-checks/src/main/java/org/sonar/flex/checks/NonEmptyCaseWithoutBreakCheck.java

```diff
@@ -49,12 +49,32 @@ public void init() {
 
   @Override
   public void visitNode(AstNode astNode) {
-    AstNode lastAstNode = astNode.getLastChild();
-    if (lastAstNode.getFirstChild().is(FlexGrammar.STATEMENT)
-      && lastAstNode.getFirstChild().getFirstChild().isNot(FlexGrammar.BREAK_STATEMENT, FlexGrammar.RETURN_STATEMENT, FlexGrammar.THROW_STATEMENT)) {
+    int jumpStmtNumber = astNode.getDescendants(FlexGrammar.BREAK_STATEMENT, FlexGrammar.RETURN_STATEMENT, FlexGrammar.THROW_STATEMENT).size();
+
+    if (jumpStmtNumber < 2) {
+      AstNode lastAstNode = astNode.getLastChild();
+      visitLastDirective(astNode, lastAstNode);
+    }
+  }
+
+  private void visitLastDirective(AstNode astNode, AstNode directive) {
+    if (isBlock(directive)) {
+      visitLastDirective(astNode, directive.getFirstChild().getFirstChild().getLastChild());
+      return;
+    }
+    if (directive.getFirstChild().is(FlexGrammar.STATEMENT)
+      && directive.getFirstChild().getFirstChild().isNot(FlexGrammar.BREAK_STATEMENT, FlexGrammar.RETURN_STATEMENT, FlexGrammar.THROW_STATEMENT)) {
       getContext().createLineViolation(this, "Last statement in this switch-clause should be an unconditional break.",
         Iterables.getLast(astNode.getChildren(FlexGrammar.CASE_LABEL)));
     }
   }
 
+  private boolean isBlock(AstNode astNode) {
+    return astNode.is(FlexGrammar.DIRECTIVE)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1067)